### PR TITLE
Diagnostics for server.env and bootstrap.properties

### PR DESCRIPTION
**Pull request by evie-lau**
Add diagnostics for clearly defined property values in server.env and bootstrap.properties (defined in `ServerPropertyValues.java`)

### CODE REVIEW

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        this.textDocumentItem = entry.getTextDocument();
+        this.hasErrors = false;
+    }
+
+    public void validateServerProperty() {
+        if (entry.isComment() || !ServerPropertyValues.canValidateKeyValues(textDocumentItem, entry.getKey())) {
+            return;
+        }
+        List<String> validValues = ServerPropertyValues.getValidValues(textDocumentItem, entry.getKey());
+        if (!validValues.stream().anyMatch(entry.getValue()::equalsIgnoreCase)) { // check if validValues contains the property value, ignoring case
```
Comment (cherylking) [inline liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java:40]: Does Liberty ignore the case for all the pre-defined values for the various properties/env variables?

File: liberty-ls/src/main/resources/DiagnosticMessages.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+INVALID_PROPERTY_VALUE=The value `{0}` is not valid for the property `{1}`.
```
Comment (cherylking) [inline liberty-ls/src/main/resources/DiagnosticMessages.properties:1]: The bootstrap.properties contains properties, but the server.env contains variables. I think we need a slightly different diagnostic messages for the two files, unless we can come up with a generic term that makes sense.

For server.env: The value `{0}` is not valid for the variable `{1}`.

**Review by cherylking — CHANGES_REQUESTED** [review]

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java (hunk 1 above)
Comment (cherylking) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java:40]: We are following up on this. Apparently some props/vars are case sensitive, and others are not. Once we have identified all the recognized values, the code can be updated for this PR.

Comment (evie-lau) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java]: Updated the code to follow the current state of case-sensitivity for the properties/variables.
In example: 
image

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        Map<String, PropertiesValidationResult> errors = new HashMap<>();
+        if (openedDocument.getUri().endsWith("bootstrap.properties") || openedDocument.getUri().endsWith("server.env")) {
+            BufferedReader br = new BufferedReader(new StringReader(text));
+            String line = null;
+            int lineNumber = 0;
+            try {
+                while ((line=br.readLine()) != null) {
+                    PropertiesValidationResult validationResult = ServerPropertyValues.validateServerProperty(line, openedDocument);
+                    if (validationResult.hasErrors()) {
+                        validationResult.setLineNumber(lineNumber);
```
Comment (cherylking) [inline liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java:52]: Does the IDE expect line number zero for the first line of the file?

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java (hunk 1 above)
Comment (cherylking) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java]: Should the test case be tweaked to cover case sensitive vs not?

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java (hunk 3 above)
Comment (evie-lau) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java:52]: Yes, it is zero-indexed, different from what the user sees in the IDE line numbers.

File: liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java (hunk 1 above)
Comment (evie-lau) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java]: Currently the language server portion of the project does not have any tests. I can look into setting up a testing framework for the project next.

Comment (cherylking) [reply liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java]: k...that is a whole we will want to fill. Go ahead and open an issue to follow up on adding tests for the LS.

**Review by cherylking — APPROVED** [review]

**Pull request merged**